src/libmongoc/src/mongoc/mongoc-topology-scanner.c

```diff
@@ -106,22 +106,28 @@ _jumpstart_other_acmds (mongoc_topology_scanner_node_t *node,
 static void
 _add_hello (mongoc_topology_scanner_t *ts)
 {
-   bson_t *cmd = &ts->hello_cmd;
    mongoc_server_api_t *api = ts->api;
 
+   BSON_APPEND_INT32 (&ts->hello_cmd, "hello", 1);
+   BSON_APPEND_BOOL (&ts->hello_cmd, "helloOk", true);
+
    if (api) {
-      BSON_APPEND_INT32 (cmd, "hello", 1);
-      _mongoc_cmd_append_server_api (cmd, api);
+      BSON_APPEND_INT32 (&ts->legacy_hello_cmd, "hello", 1);
```

Comment:
I'll see if I can work around this by correctly initialising `hello_ok` when an API version was declared so that I don't have to copy the `mongoc_server_api_t` into the server description or server monitor.

Reply:
I was able to fix this, the objects now contain the command they claim to contain.

Reply:
Well spotted. I've removed the extra fields.